### PR TITLE
elm-test upgrade

### PR DESCRIPTION
**Pull request by vikasagartha**
This fixes issue #6. 

It updates tests to latest to use the latest version of elm-community/elm-test, and installs rtfeldman/html-test-runner to display test results in the browser.

### CODE REVIEW

File: Test.elm [hunk 1]
```diff
@@ -1,11 +1,7 @@
-module Test exposing (..)
```
Comment (eeue56) [inline Test.elm:1]: why did you delete this line?

**Review by eeue56 — APPROVED** [review]

**Comment by eeue56** [comment]
One comment, but thanks a million for this!

File: Test.elm (hunk 1 above)
Comment (vikasagartha) [reply Test.elm:1]: It was causing a circular dependency, since I needed

'import Test' (its no longer ElmTest). 

I thought about changing it to 

'module FlatMatrixTests' but I figured since it was not imported anywhere, why not remove it. Is there a better solution?

Comment (eeue56) [reply Test.elm:1]: could you remain it to `Main` maybe? I prefer to be explicit where possible!

Comment (vikasagartha) [reply Test.elm:1]: done.

**Comment by roovo** [comment]
Apologies if this is a numpty question as I am v.new to elm, but I'm a a little confused why there is a run-time dependency on the test framework?

I can see why you'd need one when developing but surely users of the package don't want/need to be constrained by the version of testing framework used...

**Comment by eeue56** [comment]
@roovo the state of testing in Elm when I first made this package (during elm 0.15) was very broken. To some extent, it is still. When I get time later today to merge this PR, I'll split the tests off into their own separate thing.

**Comment by roovo** [comment]
@eeue56 - Thanks for the reply.... if it helps, and not being sure how to add the relevant changes into this PR, I've just added a new one (#8) which has moved the tests into a tests directory (with its own elm-package containing elm-test) - which I presume is what needs to be done to remove the dependency for package users.

The content of Tests.elm is pretty much as per the work done in this PR by @vikasagartha.

**Comment by eeue56** [comment]
Thanks! Taking a look now.

**Comment by eeue56** [comment]
merged, thanks, published as 3.0.1
